src/coreclr/tools/Common/TypeSystem/Common/TypeFlags.cs

```diff
@@ -55,12 +55,14 @@ public enum TypeFlags
 
         HasFinalizerComputed = 0x1000,
         HasFinalizer         = 0x2000,
```

Comment:
EagerFinalizer and CriticalFinalizer are not generic type system concepts. They do not need to have a flag here.

We only need to check these when producing EETypes. We can do the check only when building EEType.

Reply:
It is the first time I need to change something in the type system, so Iwas just following existing patterns - like HasFinalizer. 
What I hear is that instead of setting this flags we could just check for the same info that we capture in flags, but later.
Do we have a more appropriate example?

> We only need to check these when producing EETypes. We can do the check only when building EEType. 

Where is that done? At the locations where these flags are currently consumed?

Reply:
You should only need to compute these flags here: https://github.com/dotnet/runtime/blob/e3cd737cea578c629a194474fd67d660fc7a9903/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeNode.cs#L657

Something like:
```
if (type is MetadataType mdType &&
                mdType.Module == context.SystemModule &&
                mdType.Name == "WeakReference" pr "WeakReference`1"
                mdType.Namespace == "System")
{
    flags |= TypeFlags.HasEagerFinalizer;
}
```

You should not need the one in EETypeBuilderHelpers. Copying from the template type should be enough.